src/logic/taskcluster.ts

```diff
@@ -152,3 +159,89 @@ export async function retrieveTaskclusterUserCredentials(
 
   return response.json() as Promise<UserCredentials>;
 }
+
+type Action = {
+  hookId: string;
+  hookGroupId: string;
+  kind: 'hook';
+  hookPayload: unknown;
+  name: string;
+};
+
+export async function fetchActionsFromDecisionTask(
+  rootUrl: string,
+  decisionTaskId: string,
+) {
+  const url = `${rootUrl}/api/queue/v1/task/${decisionTaskId}/artifacts/public%2Factions.json`;
+
+  const response = await fetch(url);
+
+  void checkTaskclusterResponse(response);
+
+  return (await response.json()) as {
+    actions: Array<Action>;
+    variables: unknown;
+  };
+}
+
+// This function's goal is to retrigger an existing job from its jobId. It will
+// call all appropriate APIs from taskcluster and treeherder.
+export async function retrigger(retriggerJobConfig: {
+  rootUrl: string;
+  repo: string;
+  jobId: number;
+  times: number;
+}) {
+  const { rootUrl, repo, jobId, times } = retriggerJobConfig;
+  const jobInfo = await fetchJobInformationFromJobId(repo, jobId);
+
+  const { push_id: pushId } = jobInfo;
+  const decisionTaskId = await fetchDecisionTaskIdFromPushId(repo, pushId);
```

Comment:
In a follow-up, I think it would be more correct to do these calls in the RetriggerButton callback, and pass directly the `decisionTaskId` to this function.

This would be better in terms or architecture, so that the "taskcluster" module doesn't depend on the "treeherder" module; rather the caller (in this case RetriggerButton) orchestrates the calls to each of the modules.

Generally in terms of architecture, we want to avoid that 2 modules of the same layer call each other.